engine/src/redraw.cpp

```diff
@@ -1233,14 +1232,16 @@ void MCRedrawRestoreLockScreen(uint2 p_lock)
 {
 	MClockscreen = p_lock;
 	
-	MCredrawupdatescreenneeded = MClockscreen == 0 && s_screen_is_dirty && !s_screen_updates_disabled;
+	if (MClockscreen == 0 && s_screen_is_dirty && !s_screen_updates_disabled)
+        MCActionsSchedule(kMCActionsUpdateScreen);
```

Comment:
Since this particular test & call is run repeatedly from multiple places, it may make sense to factor it out.